crates/deno-subsystem/deno-model-builder/src/system_builder.rs

```diff
@@ -171,18 +225,101 @@ fn walk_module_graph(
     ) {
         match maybe_module {
             ModuleEntryRef::Module(m) => {
-                let module_source = match m {
-                    Module::Esm(e) => e.source.to_string(),
-                    Module::Json(j) => j.source.to_string(),
+                let (module_source, media_type, final_specifier, requires_rewrite) = match m {
+                    Module::Esm(e) => (
+                        e.source.to_string(),
+                        MediaType::from_specifier(specifier),
+                        specifier.clone(),
+                        false,
+                    ),
+                    Module::Json(j) => (
+                        j.source.to_string(),
+                        MediaType::from_specifier(specifier),
+                        specifier.clone(),
+                        false,
+                    ),
+                    Module::Npm(npm) => {
+                        // trigger CommonJS detection
+                        let _ = npm_loader
+                            .resolve_nv_ref(&npm.nv_reference, &PermissionsContainer::allow_all())
+                            .unwrap();
+                        let resolved = node_resolver
+                            .resolve_npm_reference(
+                                &npm.nv_reference,
+                                deno_runtime::deno_node::NodeResolutionMode::Execution,
+                                &PermissionsContainer::allow_all(),
+                            )
+                            .unwrap()
+                            .unwrap();
+
+                        match resolved {
+                            NodeResolution::BuiltIn(_) => todo!(),
```

Comment:
This probably is related to the `node` specified (as in `node::fs`):

https://docs.deno.com/runtime/manual/node/node_specifiers